internal/graph/reachableresources.go

```diff
@@ -189,83 +220,66 @@ func (crr *ConcurrentReachableResources) lookupRelationEntrypoint(
 		},
 	}
 
-	crr.scheduleChunkedRedispatch(t, reader, subjectsFilter, relationReference,
-		func(ctx context.Context, drsm dispatchableResourcesSubjectMap) error {
-			return crr.redispatchOrReport(ctx, t, relationReference, drsm, rg, entrypoint, stream, req, dispatched)
+	return crr.chunkedRedispatch(ctx, ci, reader, subjectsFilter, relationReference,
+		func(ctx context.Context, ci cursorInformation, drsm dispatchableResourcesSubjectMap) error {
+			return crr.redispatchOrReport(ctx, ci, relationReference, drsm, rg, entrypoint, stream, req, dispatched)
 		})
-	return nil
 }
 
-func min(a, b int) int {
-	if b < a {
-		return b
-	}
-	return a
-}
+var queryLimit uint64 = 100
```

Comment:
I'm not sure if this is part of this PR to address or something for a follow up, but wouldn't this hardcoded limit means that we would be potentially retrieving more elements that the LookupResources API asked for?
Say you call with limit 10, you want 10 elements at a time, but the `ReverseQueryRelationships` call is retrieving a larger page (100), more than needed.

Reply:
Correct but its necessary for efficiency reasons

Reply:
Because you're not returning these results, you're returning the results *computed* from them. As an example, say you ask for 10 resources... we might have to follow 50 resources here to compute those 10, so its better to look them up in bulk